addons/zeus/functions/fnc_playSoundZeus.sqf

```diff
@@ -24,20 +24,53 @@ private _onConfirm =
 	];
 	//Get in params again
 	_in params [["_pos",[0,0,0],[[]],3], ["_unit",objNull,[objNull]]];
+
+	// Clear sound preview
+	stopSound GVAR(soundPreview);
+	GVAR(soundPreviewSelected) = nil;
+	GVAR(soundPreview) = nil;
 	
 	// manage input on _tarets to get one list
 	_targets = [_targets] call FUNC(getListZenOwnersSelection);
 
+	_soundIDList = [];
 	if (count _targets == 0) then {
 		// play sound
-		[_pos, _range, _sound, _volume, false] call EFUNC(sounds,playSoundPos);
+		private _soundID = ([_pos, _range, _sound, _volume, false] call EFUNC(sounds,playSoundPos));
+		_soundIDList pushBack _soundID;
 	} else {
 		// send event
 		[QEGVAR(sounds,playSoundLocal), [_pos, _range, _sound, _volume, true], _targets] call CBA_fnc_targetEvent;
+		// TODO: how to get soundID from this?
```

Comment:
hmmm. I see the challenge. Would either of this be possible? 

1. local-sound function will add the soundID to the unit with broadcast so you could pick it up here? That would however potential be race-conditions depending on timing

2. Have playSoundLocal (local player) create the logic element and add to editable units for all zeus. However I am unsure how this will behave with many targets. It would spawn an logic element for each, so you can delete it per unit. Which is not bad, but if you select like 16, then I am unsure how happy the game will be 😅 

Alternatively I think its also fine to not have play-sound be stoppable if locally played. Its a one-off in any case, so minimal what issues it can cause if zeus uses the wrong sound.

Reply:
Hm - if the worst-case of a race condition is that the sound isn't stopped correctly, that's not too bad (as long as it doesn't have side effects), so 1. might be a viable option

Reply:
Only problem with 1, is that you would have to delay the current function execution until the local client has broadcasted it. and while CBA does have functions for delay execution in unscheduled, I am unsure how good that would work :P

Reply:
Probably best to leave it in this case then!